### PR TITLE
Post-build & WASM

### PR DESCRIPTION
**Pull request by anotherempty**

### CODE REVIEW

**Review by akesson — CHANGES_REQUESTED** [review]
Generally: looks good.

Three things:
- `cargo test` doesn't work (gen folder issue).
- Please add a test step in the GitHub workflow.
- There's an interesting problem regarding checksums for the Nix (which is why it's failing). Study the problem and propose a solution

**Review by akesson — CHANGES_REQUESTED** [review]
Did you do this?

> Please add a test step in the GitHub workflow.

File: crates/builder/src/generate/output.rs [hunk 1]
```diff
@@ -141,11 +136,11 @@ impl Output {
...
-            let mut checksummer = SeaHasher::new();
-            variants
-                .iter()
-                .for_each(|(_, content)| checksummer.write(content));
-            URL_SAFE.encode(checksummer.finish().to_be_bytes())
+            let contents = variants.iter().fold(Vec::new(), |mut acc, (_, contents)| {
+                acc.extend_from_slice(contents);
+                acc
+            });
+            URL_SAFE.encode(seahash::hash(&contents).to_be_bytes())
```
Comment (akesson) [inline crates/builder/src/generate/output.rs:143]: I would be very worried if this change is necessary. I.e. if seahash needs all the content in one vec. The foundation of how it works is by adding bytes to the hash (impl Hash does exactly that).

Please revert it to make sure that there is not something really weird here.

**Review by akesson — CHANGES_REQUESTED** [review]

**Comment by akesson** [comment]
I also added #17. Feel free to implement that in this PR or later.

File: crates/builder/src/config.rs [hunk 2]
```diff
@@ -101,7 +106,7 @@ impl Commands {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Serialize, Deserialize)]
 pub struct Config {
```
Comment (akesson) [inline crates/builder/src/config.rs:110]: The `metadata` field can be removed and replaced with `target_directory: Utf8PathBuf` as that is the only field that is used by the rest of the code. Like that there will be much less data serialized to file.
The `builder_deps` could be removed, no?

**Review by akesson — CHANGES_REQUESTED** [review]
I like the idea of storing the postbuild.yaml file at the prebuild stage as it simplifies the postbuild.

File: crates/builder/src/generate/output.rs (hunk 1 above)
Comment (akesson) [reply crates/builder/src/generate/output.rs]: @anotherempty this is still pending, right?

Comment (anotherempty) [reply crates/builder/src/generate/output.rs]: it's done

File: crates/builder/tests/playground.rs [hunk 3]
```diff
@@ -7,17 +7,24 @@ use fs_err as fs;
...
+    if gen.exists() {
+        fs::remove_dir_all(&gen).unwrap();
+    }
 
     cargo(&dir, ["build"]);
 
-    let out = dir.join("target").join("playground");
+    let out = dir.join("target").join("assets");
 
     insta::assert_snapshot!(out.ls_ascii(0).unwrap(), @r###"
```
Comment (akesson) [inline crates/builder/tests/playground.rs:26]: Move this to after the release build to verify that as well. And add an assert_snapshot for the wasm package as well. The `.wasm` file checksums can be redacted (i.e. replaced with something like `<redacted>`).

**Review by akesson — CHANGES_REQUESTED** [review]

**Review by akesson — CHANGES_REQUESTED** [review]
As discussed:
- test that the idea of adding deps to `[build-dependencies]` works.
- fix the issue with ordering of the lines in the test snapshot (IMO using the full relative paths and sorting before the compare is probably the easiest)
- add optional target-triple in the table header so that the wasmbindgen is only executed when building for wasm

File: crates/builder/src/prebuild/assembly.rs [hunk 4]
```diff
@@ -39,8 +41,8 @@ impl Assembly {
         for sass in &self.sass {
             log::info!("Processing sass assembly '{name}'");
             let css = sass.process(info)?;
-            let filename = sass.file.file_name().unwrap();
-            let hash = sass.out.write_file(css.as_bytes(), &site_dir, filename)?;
+            let filename = sass.file.file_name().unwrap().replace("scss", "css");
```
Comment (akesson) [inline crates/builder/src/prebuild/assembly.rs:44]: Minor thing: better to move this to the Sass class (where it is already done once) and to use the `file_stem()` with the `css` appended.

File: crates/builder/tests/common.rs [hunk 5]
```diff
@@ -61,16 +72,59 @@ impl PathExt for Utf8PathBuf {
         files.sort();
```
Comment (akesson) [inline crates/builder/tests/common.rs:72]: Do the replace before the sort because otherwise the order won't be determinstic.

**Review by akesson — CHANGES_REQUESTED** [review]
I still get an insta error because the lines for the snapshot are in a different order.

**Review by akesson — APPROVED** [review]

**Pull request merged**